x/ccv/provider/types/msg_test.go

```diff
@@ -19,3 +22,306 @@ func TestValidateConsumerId(t *testing.T) {
+				GenesisHash:                       []byte{0x01}, // cannot be empty
+				BinaryHash:                        []byte{0x01}, // cannot be empty
+				SpawnTime:                         now,
+				UnbondingPeriod:                   time.Duration(100000000000),
+				CcvTimeoutPeriod:                  time.Duration(100000000000),
+				TransferTimeoutPeriod:             time.Duration(100000000000),
+				ConsumerRedistributionFraction:    "0.75",
+				BlocksPerDistributionTransmission: 10,
+				HistoricalEntries:                 10000,
+				DistributionTransmissionChannel:   "",
+			},
+			valid: true,
+		},
+		{
+			name: "invalid - zero height",
+			params: types.ConsumerInitializationParameters{
+				InitialHeight:                     clienttypes.ZeroHeight(),
+				GenesisHash:                       []byte{0x01}, // cannot be empty
+				BinaryHash:                        []byte{0x01}, // cannot be empty
+				SpawnTime:                         now,
+				UnbondingPeriod:                   time.Duration(100000000000),
+				CcvTimeoutPeriod:                  time.Duration(100000000000),
+				TransferTimeoutPeriod:             time.Duration(100000000000),
+				ConsumerRedistributionFraction:    "0.75",
+				BlocksPerDistributionTransmission: 10,
+				HistoricalEntries:                 10000,
+				DistributionTransmissionChannel:   "",
+			},
+			valid: false,
+		},
+		{
+			name: "invalid - hash too long",
+			params: types.ConsumerInitializationParameters{
+				InitialHeight:                     clienttypes.NewHeight(3, 4),
+				GenesisHash:                       tooLongHash,
+				BinaryHash:                        []byte{0x01}, // cannot be empty
+				SpawnTime:                         now,
+				UnbondingPeriod:                   time.Duration(100000000000),
+				CcvTimeoutPeriod:                  time.Duration(100000000000),
+				TransferTimeoutPeriod:             time.Duration(100000000000),
+				ConsumerRedistributionFraction:    "0.75",
+				BlocksPerDistributionTransmission: 10,
+				HistoricalEntries:                 10000,
+				DistributionTransmissionChannel:   "",
+			},
+			valid: false,
+		},
+		{
+			name: "invalid - zero spawn time",
+			params: types.ConsumerInitializationParameters{
+				InitialHeight:                     clienttypes.NewHeight(3, 4),
+				GenesisHash:                       []byte{0x01}, // cannot be empty
+				BinaryHash:                        []byte{0x01}, // cannot be empty
+				SpawnTime:                         time.Time{},
+				UnbondingPeriod:                   time.Duration(100000000000),
+				CcvTimeoutPeriod:                  time.Duration(100000000000),
+				TransferTimeoutPeriod:             time.Duration(100000000000),
+				ConsumerRedistributionFraction:    "0.75",
+				BlocksPerDistributionTransmission: 10,
+				HistoricalEntries:                 10000,
+				DistributionTransmissionChannel:   "",
+			},
+			valid: false,
+		},
+		{
+			name: "invalid - zero duration",
```

Comment:
nit: s/zero duration/zero unbonding duration

Reply:
Is intentional as all durations are checked in the same way.